Lombiq.Hosting.Tenants.Management/Controllers/ShellSettingsEditorController.cs

```diff
@@ -62,49 +71,53 @@ public async Task<IActionResult> Edit(ShellSettingsEditorViewModel model)
+        var newTenantConfiguration = new Dictionary<string, string>();
 
         var tenantSettingsPrefix = $"{model.TenantId}Prefix:";
         var currentSettings = shellSettings.ShellConfiguration.AsEnumerable()
             .Where(item => item.Value != null &&
                 item.Key.Contains(tenantSettingsPrefix))
             .ToDictionary(key => key.Key.Replace(tenantSettingsPrefix, string.Empty), value => value.Value);
 
-        if (settingsDictionary?.Keys != null)
+        foreach (var key in tenantConfiguration.Keys)
         {
-            foreach (var key in settingsDictionary.Keys)
+            var tenantSettingsPrefixWithKey = $"{tenantSettingsPrefix}{key}";
+            if (shellSettings[key] != tenantConfiguration[key])
             {
-                var tenantSettingsPrefixWithKey = $"{tenantSettingsPrefix}{key}";
-                if (shellSettings[key] != settingsDictionary[key])
-                {
-                    newSettings[tenantSettingsPrefixWithKey] = settingsDictionary[key];
-                    newSettings[key] = settingsDictionary[key];
-                }
+                newTenantConfiguration[tenantSettingsPrefixWithKey] = tenantConfiguration[key];
+                newTenantConfiguration[key] = tenantConfiguration[key];
             }
         }
 
         var deletableKeys = currentSettings
-            .Where(item => settingsDictionary == null || !settingsDictionary.ContainsKey(item.Key))
+            .Where(item => !tenantConfiguration.ContainsKey(item.Key))
             .Select(item => item.Key);
 
         foreach (var key in deletableKeys)
         {
             var tenantSettingsPrefixWithKey = $"{tenantSettingsPrefix}{key}";
-            newSettings[key] = null;
-            newSettings[tenantSettingsPrefixWithKey] = null;
+            newTenantConfiguration[key] = null;
+            newTenantConfiguration[tenantSettingsPrefixWithKey] = null;
         }
 
-        var (locker, locked) = await _distributedLock.TryAcquireLockAsync("LOMBIQ_HOSTING_TENANTS_MANAGEMENT_SHELL_SETTINGS_EDITOR_LOCK", TimeSpan.FromSeconds(10));
+        var (locker, locked) =
+            await _distributedLock.TryAcquireLockAsync(
+                "LOMBIQ_HOSTING_TENANTS_MANAGEMENT_SHELL_SETTINGS_EDITOR_LOCK",
+                TimeSpan.FromSeconds(10));
+
         if (!locked)
         {
             throw new TimeoutException($"Failed to acquire a lock before saving settings to the tenant: {model.TenantId}.");
         }
 
         await using var acquiredLock = locker;
 
-        await _shellConfigurationSources.SaveAsync(shellSettings.Name, newSettings);
-        await _shellHost.ReloadShellContextAsync(shellSettings);
+        // We are using the shell configuration sources directly because using IShellHost.UpdateShellSettingsAsync would
+        // not save settings that has a key with multiple sections.
+        // See https://github.com/OrchardCMS/OrchardCore/issues/14481.
```

Comment:
```suggestion
        // We are using the shell configuration sources directly because using IShellHost.UpdateShellSettingsAsync would
        // not save settings that has a key with multiple sections, see
        // https://github.com/OrchardCMS/OrchardCore/issues/14481. Once this is fixed, we can get rid of the locking and
        // retrieve and save the shell settings settings with IShellHost.
```